File: app/src/main/java/com/ragdroid/dahaka/app/AppModule.java
package com.ragdroid.dahaka.app;

import com.ragdroid.dahaka.util.BaseSchedulerProvider;
import com.ragdroid.dahaka.util.SchedulerProvider;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

/**
 * Created by garimajain on 13/08/17.
 */
@Module
public abstract class AppModule {

    @Binds
    abstract BaseSchedulerProvider providerSchedulerProvider(SchedulerProvider provider);
}

File: app/src/main/java/com/ragdroid/dahaka/app/AppModule.kt
package com.ragdroid.dahaka.app

import android.app.Application
import com.bumptech.glide.Glide
import com.bumptech.glide.RequestManager
import com.ragdroid.dahaka.util.BaseSchedulerProvider
import com.ragdroid.dahaka.util.SchedulerProvider

import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by garimajain on 13/08/17.
 */
@Module
class AppModule {

    @Provides
    internal fun providerSchedulerProvider(provider: SchedulerProvider): BaseSchedulerProvider {
        return provider
    }

    @Provides
    @Singleton
    fun provideGlideRequestManager(application: Application): RequestManager = Glide.with(application)


}
